From 3dde2591f675ce65a293e50156a20938f4ef9faf Mon Sep 17 00:00:00 2001
From: Liu Ying <victor.liu@nxp.com>
Date: Fri, 31 Aug 2018 10:14:36 +0800
Subject: [PATCH 4487/5242] MLK-19409 drm/imx: dpu: crtc: Remove stream_id
 argument from dpu_crtc_get_resources()

commit  0c926f3a78eab4bc34bfed2bfedf8f560a1969e3 from
https://source.codeaurora.org/external/imx/linux-imx.git

The dpu_crtc argument of dpu_crtc_get_resources() contains the stream ID
information, so it's unnecessary to pass the stream_id argument to this
function.

Signed-off-by: Liu Ying <victor.liu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/drm/imx/dpu/dpu-crtc.c |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/gpu/drm/imx/dpu/dpu-crtc.c b/drivers/gpu/drm/imx/dpu/dpu-crtc.c
index 6fbf6f5..db2a8a0 100644
--- a/drivers/gpu/drm/imx/dpu/dpu-crtc.c
+++ b/drivers/gpu/drm/imx/dpu/dpu-crtc.c
@@ -575,10 +575,10 @@ static void dpu_crtc_put_resources(struct dpu_crtc *dpu_crtc)
 		dpu_tcon_put(dpu_crtc->tcon);
 }
 
-static int dpu_crtc_get_resources(struct dpu_crtc *dpu_crtc,
-				  unsigned int stream_id)
+static int dpu_crtc_get_resources(struct dpu_crtc *dpu_crtc)
 {
 	struct dpu_soc *dpu = dev_get_drvdata(dpu_crtc->dev->parent);
+	unsigned int stream_id = dpu_crtc->stream_id;
 	int ret;
 
 	dpu_crtc->cf = dpu_cf_get(dpu, stream_id + 4);
@@ -640,7 +640,7 @@ static int dpu_crtc_init(struct dpu_crtc *dpu_crtc,
 	if (!dpu_crtc->plane)
 		return -ENOMEM;
 
-	ret = dpu_crtc_get_resources(dpu_crtc, stream_id);
+	ret = dpu_crtc_get_resources(dpu_crtc);
 	if (ret) {
 		dev_err(dev, "getting resources failed with %d.\n", ret);
 		return ret;
-- 
1.7.9.5

